Dockerfile

```diff
@@ -13,21 +13,69 @@ LABEL name="manageiq-apache" \
 RUN yum -y install centos-release-scl-rh && \
     yum -y install --setopt=tsflags=nodocs mod_ssl && \
+    # SSSD Packages \
+    yum -y install --setopt=tsflags=nodocs sssd                         \
+                                           sssd-dbus                    \
+                                           && \
+    # Apache External Authentication Module Packages \
+    yum -y install --setopt=tsflags=nodocs mod_auth_kerb                \
+                                           mod_authnz_pam               \
+                                           mod_intercept_form_submit    \
+                                           mod_lookup_identity          \
+                                           mod_auth_mellon              \
+                                           && \
+    # IPA External Authentication Packages \
+    yum -y install --setopt=tsflags=nodocs c-ares                       \
+                                           certmonger                   \
+                                           ipa-client                   \
+                                           ipa-admintools               \
+                                           && \
+    # Active Directory External Authentication Packages \
+    yum -y install --setopt=tsflags=nodocs adcli                        \
+                                           realmd                       \
+                                           real-md                      \
+                                           oddjob                       \
+                                           oddjob-mkhomedir             \
+                                           samba-common                 \
+                                           samba-common-tools           \
+                                           && \
     yum clean all
 
+## Systemd cleanup base image
+RUN (cd /lib/systemd/system/sysinit.target.wants && for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -vf $i; done) & \
+     rm -vf /lib/systemd/system/multi-user.target.wants/* && \
+     rm -vf /etc/systemd/system/*.wants/* && \
+     rm -vf /lib/systemd/system/local-fs.target.wants/* && \
+     rm -vf /lib/systemd/system/sockets.target.wants/*udev* && \
+     rm -vf /lib/systemd/system/sockets.target.wants/*initctl* && \
+     rm -vf /lib/systemd/system/basic.target.wants/* && \
+     rm -vf /lib/systemd/system/anaconda.target.wants/*
+
 ## Remove any existing configurations
 RUN rm -f /etc/httpd/conf.d/*
 
-COPY docker-assets/entrypoint /usr/bin
+## Changing working directory to the Apache config location
+WORKDIR /etc/httpd
+
+COPY docker-assets/entrypoint              /usr/bin
+COPY docker-assets/initialize_httpd.sh     /usr/bin
 COPY docker-assets/generate_server_cert.sh /usr/bin
-COPY docker-assets/manageiq.conf /etc/httpd/conf.d/
+COPY docker-assets/manageiq.conf           /etc/httpd/conf.d/
```

Comment:
This will eventually be a config map right? (Not sure what is *not* included in this PR, since I know you have a bunch coming)